hrms/hr/doctype/shift_assignment/shift_assignment.py

```diff
@@ -128,27 +126,32 @@ def get_events(start, end, filters=None):
 		employee = ""
 		company = frappe.db.get_value("Global Defaults", None, "default_company")
 
-	conditions = get_event_conditions("Shift Assignment", filters)
-	events = add_assignments(start, end, conditions=conditions)
+	events = add_assignments(start, end, filters)
 	return events
 
 
-def add_assignments(start, end, conditions=None):
-	events = []
+def add_assignments(start, end, filters):
+	import json
 
-	query = """select name, start_date, end_date, employee_name,
-		employee, docstatus, shift_type
-		from `tabShift Assignment` where
-		(
-			start_date >= %(start_date)s
-			or end_date <=  %(end_date)s
-			or (%(start_date)s between start_date and end_date and %(end_date)s between start_date and end_date)
-		)
-		and docstatus = 1"""
-	if conditions:
-		query += conditions
+	events = []
+	if isinstance(filters, str):
+		filters = json.loads(filters)
+	filters.extend([["start_date", ">=", start], ["end_date", "<=", end], ["docstatus", "=", 1]])
+
+	records = frappe.db.get_list(
```

Comment:
```suggestion
	records = frappe.get_list(
```

You can use `frappe.get_list` directly
`frappe.db.get_list` calls get_list internally